src/platform.ts

```diff
@@ -162,10 +172,30 @@ export function getAvailablePowerShellExes(
         }
     } else {
         // Handle Linux and macOS case
+        const defaultExePath =  this.getDefaultPowerShellPath(platformDetails);
         paths.push({
-            versionName: "PowerShell Core",
-            exePath: this.getDefaultPowerShellPath(platformDetails),
+            versionName: "PowerShell Core" + (/-preview/.test(defaultExePath) ? " Preview" : ""),
+            exePath: defaultExePath,
         });
+
+        // If defaultExePath is pwsh, check to see if pwsh-preview is installed and if so, make it available.
+        // If the defaultExePath is already pwsh-preview, then pwsh is not installed - nothing to do.
+        let osExePath;
+        let osPreviewExePath;
+        if (platformDetails.operatingSystem === OperatingSystem.MacOS) {
+            osExePath = macOSExePath;
+            osPreviewExePath = macOSPreviewExePath;
+        } else if (platformDetails.operatingSystem === OperatingSystem.Linux) {
+            osExePath = linuxExePath;
+            osPreviewExePath = linuxPreviewExePath;
+        }
+
+        if ((osExePath === defaultExePath) && fs.existsSync(osPreviewExePath)) {
```

Comment:
It's super rare, but isn't it possible for them to have legacy (`powershell`) and preview (`pwsh-preview`)?

If so, this if check might have to be:

`defaultExePath !== osPreviewExePath && ...`

Reply:
I can fix this by testing for the legacy path here or maybe we should just remove support for the legacy "powershell"on Linux and macOS.  This was abandoned somewhere around the transition from 6.0.0 alpha to beta.  At this point, I'm not sure why anyone would still be running an alpha of 6.0.0 on Linux/macOS.